src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -400,6 +408,134 @@ nlohmann::json SysInfo::getNetworks() const
+        {
+            return it.at("inode") == matchN;
+            }))
+            {
+                matchInode = true;
+                inodes.push_back(matchN);
+            }
+        }
+    }
+}
+
+void parseProcFS(const char path[], nlohmann::json& JSONInput)
+{
+    struct dirent* dirent = nullptr;
+    DirPtr procDir(opendir(path));
+
+    // Open proc directory.
+    if (nullptr != procDir.get())
+    {
+        // Iterate over files in proc directory.
+        while (nullptr != (dirent = readdir(procDir.get())))
+        {
+            if ('.' != dirent->d_name[0])
+            {
+                // Proc files path (includes PID directories).
+                char pidPath[KByte] {0};
+                snprintf(pidPath, sizeof(pidPath), "%s/%s", path, dirent->d_name);
+
+                if (Utils::isNumber(dirent->d_name))
+                {
+                    struct dirent* dirent2 = nullptr;
+                    DirPtr pidDir(opendir(pidPath));
+
+                    // Open PID directory.
+                    if (nullptr == pidDir)
+                    {
+                        // Sometimes the PID folder couldn't be open due to library limitation, but it must continue.
+                        continue;
+                    }
+
+                    // Iterate over files in PID directory.
+                    std::vector<int64_t> inodes;
+                    bool matchInode{false};
+
+                    while (nullptr != (dirent2 = readdir(pidDir.get())))
+                    {
+                        if ('.' != dirent2->d_name[0])
+                        {
+                            // Pid files path (includes fd directory and stat file).
+                            char pidFilesPath[KByte] {0};
+                            snprintf(pidFilesPath, sizeof(pidFilesPath), "%s/%s", pidPath, dirent2->d_name);
+
+                            if (!strcmp(dirent2->d_name, "fd"))
+                            {
+                                struct dirent* dirent3 = nullptr;
+                                DirPtr fdDir(opendir(pidFilesPath));
+
+                                // Iterate over files in fd directory.
+                                while (nullptr != (dirent3 = readdir(fdDir.get())))
+                                {
+                                    if ('.' != dirent3->d_name[0])
+                                    {
+                                        // Fd files path.
+                                        char fdFilesPath[KByte] {0};
+                                        snprintf(fdFilesPath, sizeof(fdFilesPath), "%s/%s", pidFilesPath, dirent3->d_name);
+                                        findInodeMatch(fdFilesPath, inodes, matchInode, JSONInput);
+                                    }
+                                }
+                            }
+
+                            if (!strcmp(dirent2->d_name, "stat"))
```

Comment:
Maybe this block could be outside the `while`, because the purpose of this **if** isn't completely clear. 

I think that this also would fix the folders order dependency.

Reply:
Ditto